### ISSUE TITLE
Depth transform division by 1e4 (Question)

### ISSUE THREAD

**Issue by anargu** [xapharius/pytorch-nyuv2#8]
Hello again,

I was doing some training work with the dataset and after training my model it threw kernels filled with zero. So it was strange for me. It wasn't until I figured out the operation of division by 1e4 (at line 122) which is made to the depth tensor. I deactivated this and the model didn't end up with kernels values at zero.

But my question is more focused on to understand what does this division by 1e4 is intended to do.

Thanks in advance.

**Comment by xapharius** [comment]
Hi Anargu,

the division only converts the output of the depth images to meters, as written in the class doc. This allows for more interpretable images and also scales down the regression loss, such that it would not dominate the other tasks under an naive MTL (equal weight) approach. I would say it depends on your setup (learning rates, loss weights etc) as to why you get those zero kernels.

Hope this helps.

**Comment by anargu** [comment]
You are right, in fact, I made some changes to the class for my project case. Then I noticed your code is intended to work with NYUv2's labeled dataset and not with the raw dataset, and that is the same thing I was looking for my project (testing my model first with the labeled dataset).

My issue about the division 1e4 is due modifications I made to the code. I didn't see you were multiplying by 1e4 at the moment the code downloads depth values and write them into files (line 304). Then, on the get_item method you divide it by 1e4 (line 122).

Thanks for answering and help.
